### ISSUE TITLE
[v2 QUESTION]: The problem of replacing metric in the process of hyperparameter optimization

### ISSUE THREAD

**Issue by DengFeng-Zuo** [chemprop/chemprop#1135]
**What are you trying to do?**
In the process of hyperparameter optimization, I want to adjust the optimization direction to the minimum RMSE, maximum Pearson correlation coefficient, or maximum coefficient of determination (r2_score). But there is no relevant documentation to explain this issue, please help me.

**Screenshots**
I need to change the` val_loss` here to RMSE, Pearson correlation coefficient, and coefficient of determination(r2_score).
![que](https://github.com/user-attachments/assets/d22bcd3c-ce28-4714-a2d6-d694e02b51c8)

**Comment by KnathanM** [comment]
Try this:

1. Make a list of metrics that you want to log while searching hyperparameters and pass it in when making the Chemprop model. `metric_list = [nn.metrics.RMSE(), nn.metrics.R2Score()]` -> `models.MPNN(metrics=metric_list)`.
2. Replace "val_loss" in both the `CheckpointConfig` and `Tuneconfig` with the alias of the metric you want to track. You can find this as a class attribute of the metric. `nn.metrics.RMSE.alias` gives `rmse` and `nn.metrics.R2Score.alias` gives `r2`. 
3. Replace "min" with "max" if needed in `CheckpointConfig` and `Tuneconfig`.

We don't have the Pearson correlation coefficient defined in Chemprop, but it is available from `torchmetrics`. You can follow the instructions [here](https://chemprop.readthedocs.io/en/latest/tutorial/python/metrics.html#Custom-metrics) to adapt the `torchmetrics` version to be compatible with Chemprop. 

Let me know if that works for you.

**Comment by KnathanM** [comment]
Actually I'm remembering now that you need prepend "val/" to the alias (i.e. replace "val_loss" with "val/rmse"). See this line where we log the metric:
https://github.com/chemprop/chemprop/blob/01d6ca4164858b2122b66204f8ca934e41efd51a/chemprop/models/model.py#L198

**Comment by DengFeng-Zuo** [comment]
> Actually I'm remembering now that you need prepend "val/" to the alias (i.e. replace "val_loss" with "val/rmse"). See this line where we log the metric:
> 
> https://github.com/chemprop/chemprop/blob/01d6ca4164858b2122b66204f8ca934e41efd51a/chemprop/models/model.py#L198

Thank you very much for your help. Now the hyperparameter optimization process can proceed in the direction I need.
Specifically, the code I modified is as follows:
```

metric_list = [nn.metrics.R2Metric(),
                   nn.metrics.RMSEMetric(), 
                   nn.metrics.MAEMetric(),]
```
```
mcmpnn = multi.MulticomponentMPNN(mcmp,
                                      agg,
                                      ffn,
                                      metrics=metric_list,
                                      X_d_transform=X_d_transform,
                                      )
```
```
# Checkpoint config controls the checkpointing behavior of Ray
checkpoint_config = CheckpointConfig(
    num_to_keep=1, # number of checkpoints to keep
    checkpoint_score_attribute="val/r2", # Save the checkpoint based on this metric
    checkpoint_score_order="max", # Save the checkpoint with the lowest metric value
)
```
```
tune_config = tune.TuneConfig(
    metric="val/r2",  
    mode="max",
    num_samples=40, # number of trials to run
    scheduler=scheduler,
    search_alg=search_alg,
    trial_dirname_creator=lambda trial: str(trial.trial_id), # shorten filepaths
    
)
```

**Comment by KnathanM** [comment]
Glad that worked for you. I'll close the issue.
